packages/ui/components/Button/Button.tsx

```diff
@@ -1,31 +1,110 @@
-import classnames from 'classnames';
-import React from 'react';
-import { Size, WithColor } from '../../lib/classes';
+import React, { HTMLAttributes } from 'react';
+import styled, { css } from 'styled-components';
+import { Icon } from '@codement/ui';
 
+enum BtnType {
+  primary = 'primary',
+  secondary = 'secondary',
+  tertiary = 'tertiary',
+  transparent = 'transparent'
+}
 
-export type ButtonProps = WithColor & {
+export interface ButtonProps extends HTMLAttributes<any> {
   text?: boolean;
-  size?: Size;
-};
+  btnType?: 'primary' | 'secondary' | 'tertiary' | 'transparent'
+  disabled?: boolean;
+  size?: string;
+  icon?: string;
+  iconPosition?: 'left' | 'right'
+}
 
-export const Button: React.FunctionComponent<ButtonProps & Omit<React.ButtonHTMLAttributes<HTMLButtonElement>, 'size'>> = ({
+export const Button: React.FC<ButtonProps> = ({
+  icon,
+  iconPosition = 'left',
   children,
-  className,
-  text,
-  color,
-  size,
-  type = 'submit',
   ...props
-}) => {
-  const classes: any = {};
-  if (color) classes[`btn-${color}`] = true;
-  if (size) classes[`btn-${size}`] = true;
-  if (text) classes[`btn-${text}`] = true;
-  // eslint-disable-next-line react/button-has-type
-  return <button
-    type={type}
-    className={classnames('btn', classes, className)}
+}) => (
+
+  <StyledButton
     {...props}
-  >{children}
-  </button>;
-};
+  >
+    { icon && iconPosition === 'left' && <ButtonIcon size="medium" icon={icon} />}
+    { children }
+    { icon && iconPosition === 'right' && <ButtonIcon icon={icon} />}
+  </StyledButton>
+);
+
+const ButtonIcon = styled(Icon)`
+display:inline-block;
+`;
+
+const StyledButton = styled.button<ButtonProps>`
+  display: inline-block;
+  font-weight: 800;
+  height: 2.25rem;
+  line-height: 2.25rem;
+  padding-left: 0.75rem;
+  padding-right: 0.75rem;
+  border-radius: 0.5rem;
+  letter-spacing: 0.05em;
+  font-size: 0.875rem;
```

Comment:
Can we standardize these somehow? Also, what base are we using? Should be 10, not 16